web/src/components/pages/dashboard/challenge/achievement.ts

```diff
@@ -0,0 +1,69 @@
+export type AchievementType =
+  | {
+      type: 'invite';
+    }
+  | {
+      type: 'contribute';
+      firstContribute: boolean;
+      firstStreak: boolean;
+      hasAchieved: boolean;
+    };
+export function handleAchievements(
+  achievement: AchievementType,
+  addAchievement: any,
+  setInviteContributeAchievement: any
+): void {
+  if (achievement.type === 'invite') {
+    if (JSON.parse(sessionStorage.getItem('firstInvite'))) {
```

Comment:
Why are we using `sessionStorage` for `firstInvite`? The achievement happens immediately once we get a response from the server.

I initially imagined an interface like `handleAchievement({ type: 'firstInvite' })`. I think we're overusing `sessionStorage`, unless I'm misunderstanding.

Reply:
When a user clicks  `Copy link`, front-end will send a request to check if this is the first invite for the user. If valid, achievement toast will be triggered when the user closes the modal. So I need to store the results from the request.